File: python/prob_005.py
# prob_005.py

# Problem 5
#
# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
#
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
#

from Solver import Solver

def solve(max_val):
    answer = -1
    i = 1
    search = True
    values = range(1, max_val + 1)
    while search:
        divisible = True
        for v in values:
            if i % v != 0:
                divisible = False
                break
        if divisible:
            search = False
            answer = i
        i += 1
    return answer

def main():
    #solver = Solver(solve, 10)
    solver = Solver(solve, 20)
    solver.solve()

main()

File: src/prob_005.cpp
// prob_005.cpp

// Problem 5
// 
// 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
// 
// What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
// 

#include <iostream>

int solve(int max_val) 
{
    int s = -1;
    int i = 1;
    bool search = true;
    while(search)
    {
        bool divisible = true;
        for (int j = 1; j < max_val + 1; ++j)
        {
            if (i % j != 0)
            {
                divisible = false;
                break;
            }
        }
        if (divisible)
        {
            search = false;
            s = i;
        }
        i += 1;
    }
    return s;
}

int main()
{
    int x = solve(20);
    std::cout << x << std::endl;
    return 0;
}

